### PR TITLE
fix(VDataTable): adjust filter & sort functions to use Intl.Collator

### PR DESCRIPTION
**Pull request by Aramis13**
## Description
Implemented the comparing methods using Intl.Collator.Compare() function.
The function uses as default the locale 'en-US', Also i used a non case sensitive
compare in both functions.

## Motivation and Context
Fixes issues:
https://github.com/vuetifyjs/vuetify/issues/3672
https://github.com/vuetifyjs/vuetify/pull/4622

## How Has This Been Tested?
Tested locally with the example data in the issue https://github.com/vuetifyjs/vuetify/pull/4622.

```vue
<template>
<v-container>
  <v-card>
    <v-card-title>
      Nutrition
      <v-spacer></v-spacer>
      <v-text-field
        v-model="search"
        append-icon="search"
        label="Search"
        single-line
        hide-details
      ></v-text-field>
    </v-card-title>
    <v-data-table
      :headers="headers"
      :items="desserts"
      :search="search"
      :locale="'de'"
    >
      <template v-slot:items="props">
        <td>{{ props.item.name }}</td>
        <td class="text-xs-right">{{ props.item.calories }}</td>
        <td class="text-xs-right">{{ props.item.fat }}</td>
        <td class="text-xs-right">{{ props.item.carbs }}</td>
        <td class="text-xs-right">{{ props.item.protein }}</td>
        <td class="text-xs-right">{{ props.item.iron }}</td>
      </template>
      <template v-slot:no-results>
        <v-alert :value="true" color="error" icon="warning">
          Your search for "{{ search }}" found no results.
        </v-alert>
      </template>
    </v-data-table>
  </v-card>
</v-container>
</template>

<script>
  export default {
    created () {
      this.$vuetify.lang.current = 'es'
    },
    data () {
      return {
        search: '',
        headers: [
          {
            text: 'Dessert (100g serving)',
            value: 'name'
          },
          { text: 'Calories', value: 'calories' },
          { text: 'Fat (g)', value: 'fat' },
          { text: 'Carbs (g)', value: 'carbs' },
          { text: 'Protein (g)', value: 'protein' },
          { text: 'Iron (%)', value: 'iron' }
        ],
        desserts: [
          {
            name: 'Ämna',
            calories: 159,
            fat: 6.0,
            carbs: 24,
            protein: 4.0,
            iron: '1%'
          },
          {
            name: 'Lidia',
            calories: 237,
            fat: 9.0,
            carbs: 37,
            protein: 4.3,
            iron: '1%'
          },
          {
            name: 'Łucja',
            calories: 262,
            fat: 16.0,
            carbs: 23,
            protein: 6.0,
            iron: '7%'
          },
          {
            name: 'Robert',
            calories: 305,
            fat: 3.7,
            carbs: 67,
            protein: 4.3,
            iron: '8%'
          },
          {
            name: 'Walter',
            calories: 356,
            fat: 16.0,
            carbs: 49,
            protein: 3.9,
            iron: '16%'
          },
          {
            name: 'álter',
            calories: 375,
            fat: 0.0,
            carbs: 94,
            protein: 0.0,
            iron: '0%'
          },
          {
            name: 'Úrbano',
            calories: 392,
            fat: 0.2,
            carbs: 98,
            protein: 0,
            iron: '2%'
          },
          {
            name: 'Würst',
            calories: 408,
            fat: 3.2,
            carbs: 87,
            protein: 6.5,
            iron: '45%'
          },
          {
            name: 'Wäb',
            calories: 452,
            fat: 25.0,
            carbs: 51,
            protein: 4.9,
            iron: '22%'
          }
        ]
      }
    }
  }
</script>

```
</details>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ x] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

## Checklist:

- [ x] The PR title is no longer than 64 characters.
- [ x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and breaking changes).
- [x ] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

**Comment by johnleider** [comment]
With the release of v2, the hold has been removed but we need to get this rebased to master.

**Comment by johnleider** [comment]
I manually changed the base and it seems to be in good order, let me know if there is anything I can do to help.

**Comment by Aramis13** [comment]
Just some directions of what to do next?
Don't really know what codecov error means...

**Comment by nekosaur** [comment]
@Aramis13 You can ignore codecov for now, but you need to resolve the merge conflicts

**Comment by johnleider** [comment]
@nekosaur need final approval on this

**Comment by johnleider** [comment]
Pulse check.

File: packages/vuetify/src/components/VDataTable/VDataTable.ts [hunk 1]
```diff
@@ -88,8 +85,9 @@ export default VDataIterator.extend({
     },
     customFilter: {
       type: Function,
-      default: defaultFilter,
-    } as PropValidator<typeof defaultFilter>,
+      default: searchTableItems,
+    } as unknown as PropValidator<(items: any[], search: string | null, locale: string,
+       exclusiveHeaders: TableHeader[], nonExclusiveHeaders: TableHeader[]) => any[]>,
```
Comment (nekosaur) [inline packages/vuetify/src/components/VDataTable/VDataTable.ts]: I am not a fan of putting the end user developer in charge of handing exclusive vs non-exclusive headers. This new definition of customFilter is also a breaking change. The third argument switches from being the entire item to being the locale.

**Comment by Aramis13** [comment]
@nekosaur Hope those are the changes you mentioned. If there is something else to be done please let me know.

**Pull request merged**